### DISCUSSION TITLE
Error in deploy on VPS

### DISCUSSION THREAD

**Discussion by madhawaprera** [anasty17/mirror-leech-telegram-bot#1199] [Q&A] [answered]
When I tried to deploy This error occurs, What is the solution

Successfully built e232d70e0063
Successfully tagged seriesmirrorvps_app:latest
WARNING: Image for service app was built because it did not already exist. To re                                                                                                             build this image you must use `docker-compose build` or `docker-compose up --bui                                                                                                             ld`.
Creating seriesmirrorvps_app_1 ... done
Attaching to seriesmirrorvps_app_1
app_1  | Traceback (most recent call last):
app_1  |   File "update.py", line 39, in <module>
app_1  |     if config_dict := db.settings.config.find_one({'_id': bot_id}):  #r                                                                                                             etrun config dict (all env vars)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/collection.py",                                                                                                              line 1452, in find_one
app_1  |     for result in cursor.limit(-1):
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1248, in next
app_1  |     if len(self.__data) or self._refresh():
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1165, in _refresh
app_1  |     self.__send_message(q)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1052, in __send_message
app_1  |     response = client._run_operation(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/_csot.py", line                                                                                                              105, in csot_wrapper
app_1  |     return func(self, *args, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1330, in _run_operation
app_1  |     return self._retryable_read(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/_csot.py", line                                                                                                              105, in csot_wrapper
app_1  |     return func(self, *args, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1442, in _retryable_read
app_1  |     with self._socket_from_server(read_pref, server, session) as (sock_                                                                                                             info, read_pref):
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1282, in _socket_from_server
app_1  |     with self._get_socket(server, session) as sock_info:
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1217, in _get_socket
app_1  |     with server.get_socket(handler=err_handler) as sock_info:
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             407, in get_socket
app_1  |     sock_info = self._get_socket(handler=handler)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             520, in _get_socket
app_1  |     sock_info = self.connect(handler=handler)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             378, in connect
app_1  |     sock_info.authenticate()
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 8                                                                                                             70, in authenticate
app_1  |     auth.authenticate(creds, self)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 5                                                                                                             49, in authenticate
app_1  |     auth_func(credentials, sock_info)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 4                                                                                                             75, in _authenticate_default
app_1  |     return _authenticate_scram(credentials, sock_info, "SCRAM-SHA-1")
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 2                                                                                                             41, in _authenticate_scram
app_1  |     res = sock_info.command(source, cmd)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 7                                                                                                             67, in command
app_1  |     return command(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/network.py", lin                                                                                                             e 166, in command
app_1  |     helpers._check_command_response(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/helpers.py", lin                                                                                                             e 181, in _check_command_response
app_1  |     raise OperationFailure(errmsg, code, response, max_wire_version)
app_1  | pymongo.errors.OperationFailure: bad auth : Authentication failed., ful                                                                                                             l error: {'ok': 0, 'errmsg': 'bad auth : Authentication failed.', 'code': 8000,                                                                                                              'codeName': 'AtlasError'}
app_1  | Traceback (most recent call last):
app_1  |   File "update.py", line 39, in <module>
app_1  |     if config_dict := db.settings.config.find_one({'_id': bot_id}):  #r                                                                                                             etrun config dict (all env vars)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/collection.py",                                                                                                              line 1452, in find_one
app_1  |     for result in cursor.limit(-1):
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1248, in next
app_1  |     if len(self.__data) or self._refresh():
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1165, in _refresh
app_1  |     self.__send_message(q)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/cursor.py", line                                                                                                              1052, in __send_message
app_1  |     response = client._run_operation(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/_csot.py", line                                                                                                              105, in csot_wrapper
app_1  |     return func(self, *args, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1330, in _run_operation
app_1  |     return self._retryable_read(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/_csot.py", line                                                                                                              105, in csot_wrapper
app_1  |     return func(self, *args, **kwargs)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1442, in _retryable_read
app_1  |     with self._socket_from_server(read_pref, server, session) as (sock_                                                                                                             info, read_pref):
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1282, in _socket_from_server
app_1  |     with self._get_socket(server, session) as sock_info:
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/mongo_client.py"                                                                                                             , line 1217, in _get_socket
app_1  |     with server.get_socket(handler=err_handler) as sock_info:
app_1  |   File "/usr/lib/python3.8/contextlib.py", line 113, in __enter__
app_1  |     return next(self.gen)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             407, in get_socket
app_1  |     sock_info = self._get_socket(handler=handler)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             520, in _get_socket
app_1  |     sock_info = self.connect(handler=handler)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 1                                                                                                             378, in connect
app_1  |     sock_info.authenticate()
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 8                                                                                                             70, in authenticate
app_1  |     auth.authenticate(creds, self)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 5                                                                                                             49, in authenticate
app_1  |     auth_func(credentials, sock_info)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 4                                                                                                             75, in _authenticate_default
app_1  |     return _authenticate_scram(credentials, sock_info, "SCRAM-SHA-1")
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/auth.py", line 2                                                                                                             41, in _authenticate_scram
app_1  |     res = sock_info.command(source, cmd)
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/pool.py", line 7                                                                                                             67, in command
app_1  |     return command(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/network.py", lin                                                                                                             e 166, in command
app_1  |     helpers._check_command_response(
app_1  |   File "/usr/local/lib/python3.8/dist-packages/pymongo/helpers.py", lin                                                                                                             e 181, in _check_command_response
app_1  |     raise OperationFailure(errmsg, code, response, max_wire_version)
app_1  | pymongo.errors.OperationFailure: bad auth : Authentication failed., ful                                                                                                             l error: {'ok': 0, 'errmsg': 'bad auth : Authentication failed.', 'code': 8000,                                                                                                              'codeName': 'AtlasError'}
seriesmirrorvps_app_1 exited with code 1
![error](https://user-images.githubusercontent.com/97410960/200355894-378a589c-f722-4023-b4a7-5c427ac15e2d.JPG)

**Comment by anasty17** [comment, marked as answer]
pymongo.errors.OperationFailure: bad auth : Authentication failed., ful l error: {'ok': 0, 'errmsg': 'bad auth : Authentication failed.', 'code': 8000, 'codeName': 'AtlasError'}

READ THE ERROR MESSAGE

**Reply by madhawaprera** [reply]
Thanks! Solved the error
